app/cdap/components/NamespaceAdmin/SourceControlManagement/SourceControlManagementModal.tsx

```diff
@@ -0,0 +1,291 @@
+      missingFieldsCount++;
+      validateSuccess = false;
+    }
+    if (!formState.auth.tokenName) {
+      setTokenNameError(true);
+      missingFieldsCount++;
+      validateSuccess = false;
+    }
+    setMissingFields(missingFieldsCount);
+    setShowValidateMessage(true);
+    return validateSuccess;
+  };
+
+  const handleSubmit = () => {
+    if (!validateForm()) {
+      return;
+    }
+    onSubmit(formState);
+  };
+  return (
+    <ThemeProvider theme={AsteriskTheme}>
+      <StyledForm>
+        <BoldHeader>{T.translate(`${PREFIX}.basicHeader`)}</BoldHeader>
+        <FormGroup>
+          <Select
+            label={T.translate(`${PREFIX}.provider.label`)}
+            value={formState?.provider ? formState?.provider : providers.github}
+            onChange={(e) => {
+              handleValueChange(e, 'SET_PROVIDER');
+            }}
+            style={{ width: '100%' }}
+          >
+            {Object.values(providers).map((provider) => (
+              <MenuItem value={provider}>{provider}</MenuItem>
+            ))}
+          </Select>
+          <FormHelperText>{T.translate(`${PREFIX}.provider.helperText`)}</FormHelperText>
+          <StyledTextField
+            label={T.translate(`${PREFIX}.repoUrl.label`)}
+            value={formState?.link}
+            helperText={T.translate(`${PREFIX}.repoUrl.helperText`)}
+            onChange={(e) => {
+              handleValueChange(e, 'SET_REPOSITORY_URL');
+              setLinkError(!e.target.value);
+            }}
+            required
+            error={linkError}
+          />
+          <StyledTextField
+            label={T.translate(`${PREFIX}.branch.label`)}
+            value={formState?.defaultBranch}
+            helperText={T.translate(`${PREFIX}.branch.helperText`)}
+            onChange={(e) => {
+              handleValueChange(e, 'SET_BRANCH');
+            }}
+          />
+          <StyledTextField
+            label={T.translate(`${PREFIX}.pathPrefix.label`)}
+            value={formState?.pathPrefix}
+            helperText={T.translate(`${PREFIX}.pathPrefix.helperText`)}
+            onChange={(e) => {
+              handleValueChange(e, 'SET_PATH_PREFIX');
+            }}
+          />
+        </FormGroup>
+        <BoldHeader>{T.translate(`${PREFIX}.authHeader`)}</BoldHeader>
+        <FormLabel>{T.translate(`${PREFIX}.auth.label`)}</FormLabel>
+        <RadioGroup
+          row
+          value={formState?.auth?.type ? formState.auth.type : scmAuthType.pat}
+          onChange={(e) => {
+            handleValueChange(e, 'SET_AUTH_TYPE');
+          }}
+        >
+          <FormControlLabel
+            value={scmAuthType.pat}
+            control={<Radio />}
+            label={T.translate(`${PREFIX}.auth.pat.label`)}
+          />
+        </RadioGroup>
+        {formState?.auth?.type === scmAuthType.pat && (
+          <FormGroup>
+            <StyledTextField
+              label={T.translate(`${PREFIX}.auth.pat.tokenName`)}
```

Comment:
none of these have default text or placeholders?

Reply:
it has helperText

Reply:
helper text is a placeholder? what shows up when theres an error then?

Reply:
helperText is below the input field. on error it just turns red. But the design is subject to change